VDF.Core/FFTools/FFProbeJsonReader.cs

```diff
@@ -18,114 +18,212 @@
+
+using DSO = System.Collections.Generic.Dictionary<string, object>; 
+using LO  = System.Collections.Generic.List <object>; 
 
 namespace VDF.Core.FFTools {
+
+#if true
+// ---------------------------------------------------------------------------------
+// V1) JSON -> JsonSerializer.Deserialize -> JsonMediaInfo class -> MediaInfo class
+// ---------------------------------------------------------------------------------
+
 	static class FFProbeJsonReader {
 
-		enum JsonObjects {
-			None,
-			Streams,
-			Format
+		public class JsonMediaInfo
+		{
+			#pragma warning disable CS0649 // Field is never assigned to, and will always have its default
+			public Format? format;
+			public List<Stream>? streams;
+
+			public class Format {
+				public string? bit_rate;
+				public string? duration;
+			}
+			public class Stream	{
+				public string? bit_rate;
+				public int width;
+				public int height;
+				public string? codec_name;
+				public string? codec_long_name;
+				public string? codec_type;
+				public string? channel_layout;
+				public string? pix_fmt;
+				public string? sample_rate;
+				public int index;
+				public string? r_frame_rate;
+				public Disposition? disposition;
+			}
+			public class Disposition {
+				[JsonPropertyName("default")]
+				public int default_;
+			}
+			#pragma warning restore CS0649 // Field is never assigned to, and will always have its default
 		}
 
-		// C# no-alloc optimization that directly wraps the data section of the dll (similar to string constants)
-		// https://github.com/dotnet/roslyn/pull/24621
-		static ReadOnlySpan<byte> StreamsKeyword => new byte[] { 0x73, 0x74, 0x72, 0x65, 0x61, 0x6D, 0x73 }; // = streams
-		static ReadOnlySpan<byte> FormatKeyword => new byte[] { 0x66, 0x6F, 0x72, 0x6D, 0x61, 0x74 }; // = format
-		static ReadOnlySpan<byte> IndexKeyword => new byte[] { 0x69, 0x6E, 0x64, 0x65, 0x78 }; // = index
+		static JsonSerializerOptions serializerOptions = new JsonSerializerOptions
+		{
+			IncludeFields = true,
+		};
 
 		/// <summary>
 		/// Parses FFprobe JSON output and returns a new <see cref="MediaInfo"/>
 		/// </summary>
 		/// <param name="data">JSON output</param>
 		/// <param name="file">The file the JSON output format is about</param>
 		/// <returns><see cref="MediaInfo"/> containing information from FFprobe output</returns>
-		public static MediaInfo Read(byte[] data, string file) {
+		public static MediaInfo Read(byte[] data, string file) {;
 
-			var json = new Utf8JsonReader(data, isFinalBlock: false, state: default);
+			var json = new Utf8JsonReader(data);
+            var jsonInfo = JsonSerializer.Deserialize<JsonMediaInfo>(ref json, serializerOptions);
 
+			var info = new MediaInfo {
+				Streams = new MediaInfo.StreamInfo[jsonInfo?.streams?.Count ?? 0]
+			};
 
-			var streams = new List<Dictionary<string, object>>();
-			var format = new Dictionary<string, object>();
+			if (jsonInfo != null && jsonInfo.streams != null) {
```

Comment:
This can be merged

Reply:
Since I was programming in C# for the first time as part of these commits, optimizing a null query was not a top priority.^^
Would (jsonInfo?.streams != null) be the appropriate thing to do, or how would you merge it?

Reply:
Yes